packages/valory/skills/market_creation_manager_abci/behaviours.py

```diff
@@ -2460,32 +2314,116 @@ def get_payload(self) -> Generator[None, None, str]:
         balance = yield from self._get_balance(safe_address)
         if balance is None:
             self.context.logger.info("Couldn't get balance")
-            return CloseMarketsRound.NO_TX
+            return GetPendingQuestionsRound.NO_TX_PAYLOAD
 
         self.context.logger.info(f"Address {safe_address!r} has balance {balance}.")
         max_num_questions = min(
             len(questions), self.params.questions_to_close_batch_size
         )
         bond_required = self.params.close_question_bond * max_num_questions
+
+        # TODO uncomment
         if balance < bond_required:
             # not enough balance to close the questions
             self.context.logger.info(
                 f"Not enough balance to close {max_num_questions} questions. "
                 f"Balance {balance}, required {bond_required}"
             )
-            return CloseMarketsRound.NO_TX
+            return GetPendingQuestionsRound.NO_TX_PAYLOAD
 
-        # get the answers for those questions
-        question_to_answer = {}
+        # Prepare the Mech Requests for these questions
+        new_mech_requests = []
         for question in questions:
-            question_id = question["question"]["id"]
-            if question_id.lower() in self.shared_state.processed_question_ids:
+            question_id = question["question"]["id"].lower()
+            if question_id in self.shared_state.questions_requested_mech:
```

Comment:
Why are the `questions_requested_mech` not in the synced db?

Reply:
This was the similar approach designed previously. Since it's a single-agent service, it's easier to use the shared state in this case.